k8s_api_client/policy/api_client.rego

```diff
@@ -0,0 +1,67 @@
+package kubernetes.api.client
+
+# This information could be retrieved from the kubernetes API
+# too, but would essentially require a request per API group,
+# so for now use a lookup table for the most common resources.
```

Comment:
💭 Do you think a mechanism to use this with a resource group unknown to the rego code would be possible?

I.e. if I wanted to do something with another CRD, I'd have to copy the code, adapt it accordingly. I guess possible ways our are either what you describe there, using the API for retrieving this information; or having some extra parameter on the query functions...? 🤔 

But I suppose if we expect people to import this via copy/paste anyways (lacking package mgmt for rego libs), then it'll be fine if users with other CRDs edit this mapping.

Reply:
The same thought probably applies when considering API versions? I don't know how frequent changes are here.

Reply:
Yeah we could definitely add a function to allow providing that information - problem would be that we'd need to duplicate pretty much all of them to allow this given how we can't have optional arguments. When I said [this a few days ago](https://github.com/open-policy-agent/opa/issues/1921#issuecomment-732462439):

> I have a policy where this would greatly help simplify things by allowing the caller to provide optional arguments. 

..this was the policy I was talking about ;)

One option would be to have a single "public" function that would just take a map for input, and then dispatch to various internal functions depending on what was provided in the arguments map. That's probably the best approach to having "optional" arguments currently or what do you think?

Reply:
> One option would be to have a single "public" function that would just take a map for input, and then dispatch to various internal functions depending on what was provided in the arguments map. That's probably the best approach to having "optional" arguments currently or what do you think?

I suppose we could do something like this, too, providing "shortcut" definitions for when you want to go with the default mapping,

```rego
query_name_ns(resource, name, namespace) = query_name_ns_with_resources(resource, name, namespace, default_resource_group_mapping)
```

so anyone who needs the flexibility could use `query_name_ns_with_resources` instead of `query_name_ns`.

Reply:
Ah, yes that would work as well 👍 I think we can keep what's there now as it is, as this was more of an example demo meant to demo the concept rather than a full blown library (hence /contrib), but if others would want to pick up the torch from here they should have a good starting point here.

Reply:
Agreed! LGTM